src/Codeception/Util/JsonType.php

```diff
@@ -119,55 +119,71 @@ protected function typeComparison($data, $jsonType)
             if (!array_key_exists($key, $data)) {
                 return "Key `$key` doesn't exist in " . json_encode($data);
             }
+
             if (is_array($jsonType[$key])) {
                 $message = $this->typeComparison($data[$key], $jsonType[$key]);
+
                 if (is_string($message)) {
                     return $message;
                 }
+
                 continue;
             }
 
+            $regexMatcher = '/:regex\((.*?)\)(?:\|(boolean|integer|double|float|string|array|object|resource|resource \(closed\)|null|unknown type)|$)/';
             $regexes = [];
 
             // Match the string ':regex(' and any characters until a regex delimiter (matches 99.999% use cases) followed by character ')'
             // Place the 'any character' + delimiter matches in to an array.
-            preg_match_all('/:regex\((.*?[!"#$%&\'*+,.\/:;=?@^_`|~-])\)/', $type, $regexes);
+            preg_match_all($regexMatcher, $type, $regexes);
 
             // Do the same match as above, but replace the the 'any character' + delimiter with a place holder ($${count}).
-            $type = preg_replace_callback('/:regex\((.*?[!"#$%&\'*+,.\/:;=?@^_`|~-])\)/', function ($m) {
+            $type = preg_replace_callback($regexMatcher, function ($m) {
```

Comment:
Using different variable name instead of `$type` here and in the `preg_split` below would avoid need to refill at the bottom.

Reply:
Ah!